packages/engine/src/avatar/AvatarInputSystem.ts

```diff
@@ -121,6 +123,11 @@ export default async function AvatarInputSystem(world: World) {
     )
   }
 
+  const canvas = EngineRenderer.instance.renderer.domElement
+  canvas.addEventListener('auxclick', () => {
```

Comment:
should use `world.buttons` instead of adding event listeners

Reply:
I actually was using world.buttons originally but switched back to event listeners because I felt it'd be better than writing extra logic to detect the beginning of a click from the world.buttons mouse click.

Reply:
should be as easy as `if(world.buttons.Primary?.pressed)`